tkp/db/alchemy/image.py

```diff
@@ -0,0 +1,192 @@
+                                  ) / 2 )
+                     )
+        FROM skyregion sky
+            ,runningcatalog rc
+       WHERE sky.id = %(skyregion_id)s
+         AND rc.dataset = sky.dataset
+         AND rc.wm_decl BETWEEN sky.centre_decl - sky.xtr_radius
+                            AND sky.centre_decl + sky.xtr_radius
+         AND (  (rc.x - sky.x) * (rc.x - sky.x)
+                + (rc.y - sky.y) * (rc.y - sky.y)
+                + (rc.z - sky.z) * (rc.z - sky.z)
+             ) < %(inter_sq)s
+      ;
+    """ % {'inter_sq': inter_sq, 'skyregion_id': skyregion.id}
+    session.execute(q)
+    return inter
+
+
+def get_skyregion(session, dataset, centre_ra, centre_decl, xtr_radius):
+    """
+     gets an id for a skyregion, given a pair of central co-ordinates and a radius.
+
+     If no matching skyregion is found, a new one is inserted. In this case we also trigger execution of
+     `updateSkyRgnMembers` for the new skyregion - this performs a simple assocation with current members of the
+     runningcatalog to find sources that should be visible in the new skyregion, and updates the assocskyrgn table
+     accordingly.
+
+     args:
+        session (sqlalchemy.orm.session.Session): a SQLAlchemy session object
+        dataset_id (int): the dataset ID
+        centre_ra (float): center RA coordinate
+        centre_decl (float): center DECL coordinate
+        xtr_radius (float): The extraction radius
+
+     returns:
+        tkp.db.models.Skyregion: a SQLalchemy skyregion
+     """
+    skyregion = session.query(Skyregion).filter(Skyregion.dataset == dataset,
+                                                Skyregion.centre_ra == centre_ra,
+                                                Skyregion.centre_decl == centre_decl,
+                                                Skyregion.xtr_radius == xtr_radius).first()
+    if not skyregion:
+        x, y, z = eq_to_cart(centre_ra, centre_decl)
+        skyregion = Skyregion(dataset=dataset, centre_ra=centre_ra, centre_decl=centre_decl,
+                              xtr_radius=xtr_radius, x=x, y=y, z=z)
+        session.add(skyregion)
+        session.flush()
+        update_skyregion_members(session, skyregion)
+    return skyregion
+
+
+def insert_image(session, dataset, freq_eff, freq_bw, taustart_ts, tau_time, beam_smaj_pix, beam_smin_pix,
+                 beam_pa_rad, deltax, deltay, url, centre_ra, centre_decl, xtr_radius, rms_qc, freq_bw_max=0.0,
+                 rms_min=None, rms_max=None, detection_thresh=None, analysis_thresh=None):
+    """
+    Insert an image for a given dataset.
+
+    Args:
+        session (sqlalchemy.orm.session.Session): A SQLalchemy sessions
+        dataset (int): ID of parent dataset.
+        freq_eff: See :ref:`Image table definitions <schema-image>`.
+        freq_bw: See :ref:`Image table definitions <schema-image>`.
+        freq_bw_max (float): Optional override for freq_bw, not used if 0.
```

Comment:
What was the reasoning behind adding a freq_bw_max field to the image? Why not just use whatever's currently in the job_config?